plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/variation-selector/attribute-options/frontend.ts

```diff
@@ -1,9 +1,152 @@
+import { store, getContext, getElement } from '@wordpress/interactivity';
 
-store( 'woocommerce/add-to-cart-with-options', {
-	state: {},
-	actions: {},
-} );
+type Option = {
+	value: string;
+	label: string;
+	isSelected: boolean;
+};
+
+type Context = {
+	selected: string | null;
+	option: Option;
+	options: Option[];
+};
+
+type PillsContext = Context & {
+	tabIndex?: number;
+	selected?: string;
+	focused?: string;
+};
+
+store(
+	'woocommerce/add-to-cart-with-options-variation-selector-attribute-options__pills',
+	{
+		state: {},
+		actions: {
+			handleClick() {
+				const context = getContext< PillsContext >();
+				if ( context.selected === context.option.value ) {
+					context.selected = '';
+				} else {
+					context.selected = context.option.value;
+				}
+				context.focused = context.option.value;
+			},
+			handleKeyDown( event: KeyboardEvent ) {
+				const context = getContext< PillsContext >();
+
+				let keyWasProcessed = false;
+
+				switch ( event.key ) {
+					case ' ':
+						if ( context.selected === context.option.value ) {
+							context.selected = '';
+						} else {
+							context.selected = context.option.value;
+						}
+						context.focused = context.option.value;
+						keyWasProcessed = true;
+						break;
+
+					case 'Up':
+					case 'ArrowUp':
+					case 'Left':
+					case 'ArrowLeft': {
+						const index = context.options.findIndex(
+							( option ) => option.value === context.option.value
+						);
+						if ( index === -1 ) return;
+						const at =
+							index > 0 ? index - 1 : context.options.length - 1;
+
+						context.selected = context.options[ at ].value;
+						context.focused = context.selected;
+						keyWasProcessed = true;
+						break;
+					}
+
+					case 'Down':
+					case 'ArrowDown':
+					case 'Right':
+					case 'ArrowRight': {
+						const index = context.options.findIndex(
+							( option ) => option.value === context.option.value
+						);
+						if ( index === -1 ) return;
+						const at =
+							index < context.options.length - 1 ? index + 1 : 0;
+
+						context.selected = context.options[ at ].value;
+						context.focused = context.selected;
+						keyWasProcessed = true;
+						break;
+					}
+					default:
+						break;
+				}
+
+				if ( keyWasProcessed ) {
+					event.stopPropagation();
+					event.preventDefault();
+				}
+			},
+		},
+		callbacks: {
+			watchSelected() {
+				const context = getContext< PillsContext >();
+
+				if ( ! context.selected && ! context.focused ) {
+					if ( context.options[ 0 ] === context.option ) {
+						context.tabIndex = 0;
+					}
+					return;
+				}
+
+				context.option.isSelected =
+					context.selected === context.option.value;
+
+				if (
+					context.option.isSelected ||
+					context.focused === context.option.value
+				) {
+					context.tabIndex = 0;
+					if ( context.focused ) {
+						const { ref } = getElement();
+						ref?.focus();
+					}
+					return;
+				}
+
+				context.tabIndex = -1;
+			},
+		},
+	}
```

Comment:
Let's make these stores private.

```suggestion
	}, 
    { lock: true }
```

Reply:
done -> [c1a8673](https://github.com/woocommerce/woocommerce/pull/56140/commits/c1a8673d37a80537f79e11fa8b1e3ed2b1c9e4ca)